### DISCUSSION TITLE
Error: Tokens are invalid

### DISCUSSION THREAD

**Discussion by ianmurrays** [teslamate-org/teslamate#3213] [Q&A]
Hi all,
my teslamate instance somehow got logged out and started asking for an Access and Refresh tokens. I'm providing it with new ones (and verified working through cURL), but I keep getting the "Tokens are invalid" error. Logs show:

```
[error] POST https://auth.tesla.com/oauth2/v3/token -> 403
```

A curl request like this one (from the same server of course):

```bash
curl -H "Authorization: Bearer $TK" https://owner-api.teslamotors.com/api/1/vehicles
```

works fine.

I've tried restarting the teslamate service, the whole server, nothing seems to work. Anything I'm missing? I'm on Teslamate 1.27.2

**Comment by ianmurrays** [comment]
I've also tried routing the requests through a different IP (because it would seem that my server IP was banned by Akamai), but still no dice. Anyone else having issues with running teslamate on Digitalocean servers?

**Reply by samerfahmy** [reply]
I am also seeing the same issue with my Digitalocean setup.

**Reply by bodosom** [reply]
> Anyone else having issues with running teslamate on Digitalocean servers?

Yes, including errors from Akamai. I'm not having any problems with non-Digital Ocean hosts.

**Reply by cwanja** [reply]
Open a ticket with them and see what they can investigate. It could be they blocked the domain (too many requests) or Tesla did (for same reason). 

With GCP and non-DigitalOcean instances working, seems unlikely a TeslaMate issue.

**Reply by samerfahmy** [reply]
I just tried moving to GCP, and getting hit with the same issue.
Locally the call works, but on the GCP server it fails.

**Comment by BabyYeggie** [comment]
Local server on ISP seems to have this issue but GCP process seems to be running ok.

**Reply by samerfahmy** [reply]
I just tried moving to GCP, and getting hit with the same issue.
I wonder if it is region dependent. Which region are you running your GCP instance in?

**Reply by BabyYeggie** [reply]
I'm running mine in US West. It seems to be having issues with the size of my postgres database (910MB).

**Reply by roelsroels** [reply]
Same issue for my AMS2 region setup.. now moving the setup to my local home env... kinda anoying.

**Comment by bodosom** [comment]
I checked a couple of Digital Ocean data centers and neither are working. Linode/Akamai, AWS and NetActuate (in the northeast US) are working as is my ISP.

**Comment by dominikvpb** [comment]
Same issue on Digital Ocean / AMS datacenter.
I moved my install to my home server now. Works again. It seems the issues started around 22nd of May. That was at least the cutoff date from that I didn't receive any data anymore.

**Comment by LadPack** [comment]
I am having an extremely similar issue on a homeserver,  the output of the log is.

I am able to ping google as well as the auth.tesla.com and I just updated my docker to the current version, any ideas?

`{"log":"2024-03-20 22:31:24.918 [info] GET /\n","stream":"stdout","time":"2024-03-20T22:31:24.919136293Z"}

{"log":"2024-03-20 22:31:24.920 [info] Sent 302 in 1ms\n","stream":"stdout","time":"2024-03-20T22:31:24.920428236Z"}

{"log":"2024-03-20 22:31:24.925 [info] GET /sign_in\n","stream":"stdout","time":"2024-03-20T22:31:24.925839243Z"}

{"log":"2024-03-20 22:31:24.927 [info] Sent 200 in 2ms\n","stream":"stdout","time":"2024-03-20T22:31:24.927325186Z"}

{"log":"2024-03-20 22:31:43.029 [error] POST https://auth.tesla.com/oauth2/v3/token -\u003e error: \"non-existing domain\" (8001.748 ms)\n","stream":"stdout","time":"2024-03-20T22:31:43.>

{"log":"2024-03-20 22:32:17.501 [error] GET https://api.github.com/repos/teslamate-org/teslamate/releases/latest -\u003e error: \"non-existing domain\" (8000.923 ms)\n","stream":"stdout">

{"log":"2024-03-20 22:32:17.501 [warning] Update check failed: \"non-existing domain\"\n","stream":"stdout","time":"2024-03-20T22:32:17.502121942Z"}

{"log":"2024-03-20 22:34:39.499 [warning] Cannot refresh access token: :not_signed_in\n","stream":"stdout","time":"2024-03-20T22:34:39.499775561Z"}

{"log":"2024-03-20 22:41:28.428 [info] GET /\n","stream":"stdout","time":"2024-03-20T22:41:28.429206233Z"}

{"log":"2024-03-20 22:41:28.430 [info] Sent 302 in 1ms\n","stream":"stdout","time":"2024-03-20T22:41:28.430346127Z"}

{"log":"2024-03-20 22:41:28.436 [info] GET /sign_in\n","stream":"stdout","time":"2024-03-20T22:41:28.436804982Z"}

{"log":"2024-03-20 22:41:28.438 [info] Sent 200 in 1ms\n","stream":"stdout","time":"2024-03-20T22:41:28.438637109Z"}

{"log":"2024-03-20 22:41:45.136 [error] POST https://auth.tesla.com/oauth2/v3/token -\u003e error: \"non-existing domain\" (8001.655 ms)\n","stream":"stdout","time":"2024-03-20T22:41:45.`

**Reply by brianmay** [reply]
Looks like some sort of DNS issue on your system? "non-existing domain"

**Reply by lits101** [reply]
I'm having this exact issue. "non-existing domain" for both api.github.com and x.tesla.com.

**Comment by cwanja** [comment]
Breaking out your issue to this post @LadPack.

> My docker installation is hosted on a proxmox lxc running Debian. Yesterday I tried changing my DNS from 8.8.8.8 to 1.1.1.1 and that did not work. Also tried a few different reboots including the whole server

Google and Cloudflare. What is the default DNS, your home or a datacenter? This seems local to your installation and that is why I am trying to narrow down options. Do you have any custom Docker network instructions?

**Reply by LadPack** [reply]
Primary DNS per my router is 68.94.156.11, I am running the basic docker compose just with the password changed.

I have been using the 8.8.8.8 DNS for most of my LXC, and it was working earlier. I am trying to change to 68.94.156.11

**Comment by Joshfindit** [comment]
I'd suggest adding some documentation to the error message. I just got hit by this as spinning up a DO droplet is simple and I I was chasing `401` token issues instead of thinking they would return 401 to denied IP ranges.

Something like this would be helpful:
```
teslamate_1  | 2025-01-10 19:08:32.230 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/### -> 401 (100.411 ms)
teslamate_1  | 2025-01-10 19:08:32.232 car_id=1 [error] Error / not_signed_in (could also be blocked from this IP)
```

**Comment by livingfree2023** [comment]
I have been using an Oracle Free-Tier cloud instance in Tokyo for many years. Never ever had any issue but my teslamate was logged out on 8th Jan 2025, thus losing all the data between 8th and today. 

I couldn't re-login with new tokens. Then I decided to backup, recreate the instances to the latest images then restore everything. It worked so far. I don't know what caused it. Any ideas?
